lib/manageiq/util/file_splitter.rb

```diff
@@ -0,0 +1,190 @@
+    @input_file     = options[:input_file] || ARGF
+    @input_filename = options[:input_filename]
+    @byte_count     = options[:byte_count] || 10_485_760
+    @position       = 0
+
+    setup_logging(options)
+    setup_ftp(options)
+  end
+
+  def split
+    until input_file.eof?
+      if ftp
+        split_ftp
+      else
+        split_local
+      end
+      @position += byte_count
+    end
+  ensure
+    input_file.close
+    ftp.close if ftp
+  end
+
+  private
+
+  def setup_logging(options)
+    self.class.instance_logger = Logger.new(STDOUT) if options[:verbose]
+  end
+
+  def setup_ftp(options)
+    if options[:ftp_host]
+      @uri      = options[:ftp_host]
+      @ftp_user = options[:ftp_user] || ENV["FTP_USERNAME"] || "anonymous"
+      @ftp_pass = options[:ftp_pass] || ENV["FTP_PASSWORD"]
+      @ftp      = connect
+
+      @input_filename = File.join(options[:ftp_dir] || "", File.basename(input_filename))
+    end
+  end
+
+  def login_credentials
+    [@ftp_user, @ftp_pass]
+  end
+
+  def split_local
+    File.open(next_split_filename, "w") do |split_file|
+      split_file << input_file.read(byte_count)
+    end
+  end
+
+  # Specific version of Net::FTP#storbinary that doesn't use an existing local
+  # file, and only uploads a specific size from the input_file
+  FTP_CHUNKSIZE = ::Net::FTP::DEFAULT_BLOCKSIZE
+  def split_ftp
+    ftp_mkdir_p
+    ftp.synchronize do
+      ftp.send(:with_binary, true) do
+        conn         = ftp.send(:transfercmd, "STOR #{next_split_filename}")
+        buf_left     = byte_count
+        while buf_left.positive?
+          cur_readsize = buf_left - FTP_CHUNKSIZE >= 0 ? FTP_CHUNKSIZE : buf_left
+          buf = input_file.read(cur_readsize)
+          break if buf == nil # rubocop:disable Style/NilComparison (from original)
+          conn.write(buf)
+          buf_left -= FTP_CHUNKSIZE
+        end
+        conn.close
+        ftp.send(:voidresp)
+      end
+    end
+  rescue Errno::EPIPE
+    # EPIPE, in this case, means that the data connection was unexpectedly
+    # terminated.  Rather than just raising EPIPE to the caller, check the
+    # response on the control connection.  If getresp doesn't raise a more
+    # appropriate exception, re-raise the original exception.
+    getresp
+    raise
+  end
+
+  def ftp_mkdir_p
+    dir_path = File.dirname(input_filename)[1..-1].split('/') - ftp.pwd[1..-1].split("/")
+    create_directory_structure(dir_path.join('/'))
+  end
+
+  def input_filename
+    @input_filename ||= File.expand_path(input_file.path)
+  end
+
+  def next_split_filename
+    "#{input_filename}.#{'%05d' % (@position / byte_count + 1)}"
+  end
+
+  def self.parse_byte_value(bytes)
+    match = bytes.match(/^(?<BYTE_NUM>\d+)(?<BYTE_QUALIFIER>M|K)?$/i)
+    raise ArgumentError, "Invalid byte-count", [] if match.nil?
+
+    bytes = match[:BYTE_NUM].to_i
+    if match[:BYTE_QUALIFIER]
+      bytes *= { "k" => 1024, "m" => 1_048_576 }[match[:BYTE_QUALIFIER].downcase]
```

Comment:
I think there's a method in ActiveSupport that does this for you.

Reply:
Please see https://github.com/ManageIQ/manageiq/pull/17549

Reply:
For posterity's sake, I don't think it does.